### PR TITLE
Gather comments from headers

### PR DESCRIPTION
**Pull request by CGMossa**
Adding these lines to the `clang`-call enables getting some comments.

This is not much, but it is *something*. Maybe it will be more useful later

### CODE REVIEW

**Comment by CGMossa** [comment]
This is ready for review. I just don't know who cares enough to approve / disapprove of this.

**Comment by multimeric** [comment]
Ooh, I like this. Two points of feedback. One, I note that almost all of the doc comments have a leading space that at least visually looks annoying. If rustdoc is not stripping out this space then it's probably annoying enough to fix. The other thing is just that I'd like to see an extendr build run using these bindings just to run it through the tests and linting etc. Otherwise it looks good and I think it would be valuable for sure.

**Comment by CGMossa** [comment]
There is an option to use `rustfmt` for the generated binding. Issue is it is completely broken for these headers.
Instead, I opted to use an instrospection thing, called `ParseCallbacks`, that allows one to edit what bindgen does in a trait-way.
I implemented a parse_comment-method;

I have investigated this to see if we had something for #139, but this parse-callback just provides you with this information as it goes through, the name of the macro, and the tokens of the macro, and you ought to do something else with it. There is no return nothing. But that's an aside.

**Comment by CGMossa** [comment]
For the other point, see https://github.com/extendr/extendr/pull/497, once it finishes whatever it needs to do.

**Comment by CGMossa** [comment]
Okay, so the answer is, that this `[redacted].clang_arg("-fretain-comments-from-system-headers")` is the problem.
I don't know why, and I cannot check this with a Mac machine.
I guess it could be a todo for someone with a Mac.

**Comment by CGMossa** [comment]
Alright; I'm re-running everything, but it already worked out, and I've addressed comments from you @multimeric.
PS: I wrote this message on the wrong PR.
But this one is ready for round two.

**Review by multimeric — APPROVED** [review]
Looks great!

File: build.rs [hunk 1]
```diff
@@ -508,18 +509,24 @@ fn generate_bindings(r_paths: &InstallationPaths, version_info: &RVersionInfo) {
 
     // Finish the builder and generate the bindings.
     let bindings = bindgen_builder
+        .generate_comments(true)
+        .parse_callbacks(Box::new(RCallbacks))
+        .clang_arg("-fparse-all-comments")
+        // .clang_arg("-fretain-comments-from-system-headers")
```
Comment (yutannihilation) [inline build.rs:515]: Do we need this line?

```suggestion
```

**Review by yutannihilation — APPROVED** [review]
Looks good to me!

**Comment by CGMossa** [comment]
No, but https://github.com/extendr/libR-sys/pull/141#issuecomment-1448804819

**Comment by CGMossa** [comment]
It didn't do anything on the windows part.
But it clearly affects the other platforms. So there could be something
there...

I'll remove it (or either of you can remove the line) and then let's merge
this.
It is not clean to have commented lines or code anyways.

**Comment by multimeric** [comment]
Urgh I just remembered the changelog. Please add a PR updating it too.

**Comment by yutannihilation** [comment]
Thanks! Putting it behind `#[cfg(not(target_os = "macos"))]` can be an option, but I'm fine with either.

**Pull request merged**